### ISSUE TITLE
Support resume experiment better

### ISSUE THREAD

**Issue by squirrelsc** [microsoft/nni#1463]
Currently, if the nnictl is stopped due to reboot computer or other reasons, the running trials cannot be resumed, even the trials run on remote platforms. It should be able to get them back, so that the trials don't need to be rerun. 

1. The running trials should be able to resend intermediate results, if server doesn't receive them.
2. The completed trials may be recycled by remote platform. If so, they should be able to rerun automatically.

**Comment by ultmaster** [comment]
I agree. Getting trials back is very useful, if the machine on which you run nni is unstable and sometimes reboots, for example, Windows.

In my case, it's a problematic driver.
